pkg/store/storepb/custom.go

```diff
@@ -517,3 +517,27 @@ func (c *SeriesStatsCounter) Count(series *Series) {
 		}
 	}
 }
+
+func (m *SeriesRequest) ToPromQL() string {
+	return m.QueryHints.toPromQL(m.Matchers)
+}
+
+// IsSafeToExecute returns true if the function or aggregation from the query hint
+// can be safely executed by the underlying Prometheus instance without affecting the
+// result of the query.
+func (m *QueryHints) IsSafeToExecute() bool {
+	distributiveOperations := []string{
+		"max",
+		"max_over_time",
+		"min",
+		"min_over_time",
```

Comment:
May I ask why `avg_over_time` is not safe? Is it because in Thanos we do deduplication at Query time?
If we do write time deduplication like Cortex then this seems fine

Reply:
Yes, that's the reason. I believe it should be safe if you have unique data.